app/webpacker/components/Results/Rankings/index.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Container } from 'semantic-ui-react';
+import { useQuery } from '@tanstack/react-query';
+import RankingsTable from './RankingsTable';
+import WCAQueryClientProvider from '../../../lib/providers/WCAQueryClientProvider';
+import { getRankings } from '../api/rankings';
+import Loading from '../../Requests/Loading';
+import { rankingsUrl } from '../../../lib/requests/routes.js.erb';
+import ResultsFilter from '../resultsFilter';
+
+export default function Wrapper({
+  event, region, year, rankingType, gender, show,
+}) {
+  return (
+    <WCAQueryClientProvider>
+      <Rankings
+        initialEvent={event}
+        initialRegion={region}
+        initialYear={year}
+        initialRankingType={rankingType}
+        initialGender={gender}
+        initialShow={show}
+      />
+    </WCAQueryClientProvider>
+  );
+}
+
+export function Rankings({
+  initialEvent, initialRegion, initialRankingType, initialGender, initialShow,
+}) {
+  const [event, setEvent] = useState(initialEvent);
+  const [region, setRegion] = useState(initialRegion ?? 'all');
+  const [rankingType, setRankingType] = useState(initialRankingType);
+  const [gender, setGender] = useState(initialGender);
+  const [show, setShow] = useState(initialShow ?? 'Persons');
```

Comment:
This feels a bit off. Generally the only time you might want to pass an initial state and then create a new independent state is when there's some kind of form where the user can change things, I think.

Ideally, either the back-end is providing the (up-to-date) state, removing the need for these `useState`s, or the front-end is handling everything, removing the need for the `initial...` proprs in `Rankings`.

Reply:
Well the backend is providing the data read from the URL. We could parse the URL from the frontend I guess

Reply:
I think it's worth it for the consistency, yeah.